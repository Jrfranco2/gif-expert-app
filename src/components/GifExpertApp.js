import { useState } from "react";
import AddCategory from "./AddCategory";
import GifGrid from "./GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);
  return (
    <div>
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ul>
        {categories.map((el) => (
          <GifGrid key={el} category={el} />
        ))}
      </ul>
    </div>
  );
};

export default GifExpertApp;
