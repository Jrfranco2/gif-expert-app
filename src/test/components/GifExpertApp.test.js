import { shallow } from "enzyme";
import GifExpertApp from "../../components/GifExpertApp";

describe("Pruebas en <GifExpertApp/>", () => {
  test("debe hacer match con el snapshot", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar una lista de categorias", () => {
    const categories = ["One Punch", "Dragon ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
