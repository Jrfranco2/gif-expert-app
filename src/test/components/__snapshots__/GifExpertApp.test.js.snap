// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ul>
</div>
`;

exports[`Pruebas en <GifExpertApp/> debe hacer match con el snapshot 1`] = `
<div>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</div>
`;
